resolwe_bio/tools/expressionmerge_archive.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+# pylint: disable=missing-docstring,invalid-name
+"""Merge columns of multiple experiments by gene id."""
+from __future__ import absolute_import
+import argparse
+import csv
+import os
+import pandas as pd
+import numpy as np
+
+import utils
+
+parser = argparse.ArgumentParser(description="Merge columns of multiple experiments by gene id.")
+
+parser.add_argument('-f', '--input_file', required=True, help="Txt file with paths to expression files.")
+
+args = parser.parse_args()
+
+
+def merge_expressions(input_file):
+
+    df = pd.DataFrame(np.nan, index=[], columns=[])
+    headers = []
+
+    with open(input_file, 'r') as tfile:
+        for line in tfile:
+            line = line.rstrip()
+            header = os.path.basename(line).rstrip()
+            if header.endswith('.tab.gz'):
+                header = header[:-7]
+            headers.append(header)
+
+            with utils.gzopen(line) as csvfile:
+                reader = pd.read_csv(csvfile, index_col='Gene', delimiter='\t', dtype=str)
+                df = pd.concat([df, reader], axis=1)
+    df.columns = headers
+    return df
+
+
+def get_output_file_name(input_file):
+    with open(input_file, 'r') as tfile:
+        first_line = tfile.readline().strip()
+        dir_name = os.path.split(os.path.dirname(os.path.dirname(first_line)))[0]
```

Comment:
Provide some comments what are you doing here. Why all theese splits as dirnames? Or try to pick more descriptive names.